### PR TITLE
Add more http.client_ip cases

### PR DESCRIPTION
**Pull request by smola**
## Description

* Check that client_ip and associated tags are in the root span (not _any_ span).
* Check presence of all possible headers (Fastly and CloudFlare headers are tested separately, since support is less consistent across languages).
* Check correct resolution for all header names.

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [x] If this PR modifies anything else than strictly the default scenario, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/system-tests-ci.md)). 
* [x] CI is green
   * [x] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com

### CODE REVIEW

File: tests/test_standard_tags.py [hunk 1]
```diff
@@ -172,64 +172,118 @@ def test_route(self):
...
+    def setup_client_ip_with_appsec_event_and_vendor_headers(self):
+        self._setup_with_attack()
+
+    @missing_feature(library="java", reason="missing fastly-client-ip, cf-connecting-ip, cf-connecting-ipv6")
+    @missing_feature(library="golang", reason="missing fastly-client-ip, cf-connecting-ip, cf-connecting-ipv6")
+    @missing_feature(library="nodejs", reason="missing fastly-client-ip, cf-connecting-ip, cf-connecting-ipv6")
+    @missing_feature(library="ruby", reason="missing fastly-client-ip, cf-connecting-ip, cf-connecting-ipv6")
+    @missing_feature(library="php", reason="missing fastly-client-ip, cf-connecting-ip, cf-connecting-ipv6")
+    def test_client_ip_with_appsec_event_and_vendor_headers(self):
+        """Test that meta tag are correctly filled when an appsec event is present and ASM is enabled, with vendor headers"""
```
[the remark this answers is not in the document]
Comment (cbeauchesne) [reply tests/test_standard_tags.py:277]: @smola do you have a RFC link for this ?

[the remark this answers is not in the document]
Comment (smola) [reply tests/test_standard_tags.py:277]: @simon-id @cbeauchesne The link is at the class level. These vendor headers were added in a subsequent update

[the remark this answers is not in the document]
Comment (simon-id) [reply tests/test_standard_tags.py:277]: I know we're supposed to use these new headers for sourcing the client IP, but i don't think it said anywhere that we should add them all as tags in `http.request.headers....` that's a different list no ?

[the remark this answers is not in the document]
Comment (smola) [reply tests/test_standard_tags.py:277]: I don't think there's a separate list? And I'm not sure what would be the rationale for that. cc @robertpi what do you think?

**Review by cbeauchesne — APPROVED** [review]

**Comment by cbeauchesne** [comment]
@smola Any ambition to merge this PR?  :)

**Comment by smola** [comment]
@cbeauchesne Yep. I solved conflicts now. But I'm on hold as I resolve some offline question about the spec.

File: tests/test_standard_tags.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (smola) [reply tests/test_standard_tags.py:327]: @simon-id There was, indeed, a separate list, and it did not include the new IP headers. This was just probably an overlook, and after discussing with @robertpi, I have updated the spec to make the list of collected headers a superset of headers used for IP resolution.

**Pull request merged**